mixxx does not build on unstable after an upgrade

## Issue description
When I try to build , I get:
```
In file included from src/library/trackcollection.cpp:7:0:
/nix/store/zbv02z6amzm90k874288cvm182ifvaf3-sqlite-3.19.3-dev/include/sqlite3.h:3712:16: error: using typedef-name 'sqlite3_value' after 'struct'
 typedef struct sqlite3_value sqlite3_value;
                ^
In file included from src/library/trackcollection.cpp:4:0:
src/library/trackcollection.h:38:20: note: 'sqlite3_value' has a previous declaration here
 typedef struct Mem sqlite3_value;
                    ^
In file included from src/library/trackcollection.cpp:7:0:
/nix/store/zbv02z6amzm90k874288cvm182ifvaf3-sqlite-3.19.3-dev/include/sqlite3.h:3712:30: error: conflicting declaration 'typedef int sqlite3_value'
 typedef struct sqlite3_value sqlite3_value;
                              ^
In file included from src/library/trackcollection.cpp:4:0:
src/library/trackcollection.h:38:20: note: previous declaration as 'typedef struct Mem sqlite3_value'
 typedef struct Mem sqlite3_value;
                    ^
scons: *** [lin64_build/library/trackcollection.o] Error 1
scons: building terminated because of errors.
builder for ‘/nix/store/db0rqqylzxnz682m5r218ix187d43x5y-mixxx-2.0.0.drv’ failed with exit code 2
error: build of ‘/nix/store/db0rqqylzxnz682m5r218ix187d43x5y-mixxx-2.0.0.drv’ failed
```

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
17.09pre111447.a7c8f5e419 (Hummingbird)
* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.11.13
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
"17.09pre111447.a7c8f5e419"
* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`)
build-use-sandbox = true

@aszlig @cillianderoiste Any ideas?

--- a/pkgs/applications/audio/mixxx/default.nix
+++ b/pkgs/applications/audio/mixxx/default.nix
@@ -13,6 +13,8 @@ stdenv.mkDerivation rec {
     sha256 = "0vb71w1yq0xwwsclrn2jj9bk8w4n14rfv5c0aw46c11mp8xz7f71";
   };
 
+  patches = [ ./sqlite.patch ]; # from: https://bugs.gentoo.org/show_bug.cgi?id=622776
+
   buildInputs = [
     chromaprint fftw flac libid3tag libmad libopus libshout libsndfile
     libusb1 libvorbis pkgconfig portaudio portmidi protobuf qt4

--- a/pkgs/applications/audio/mixxx/sqlite.patch
+++ b/pkgs/applications/audio/mixxx/sqlite.patch
@@ -1,0 +1,13 @@
+diff -urN old/src/library/trackcollection.h mixxx-2.0.0/src/library/trackcollection.h
+--- old/src/library/trackcollection.h	2017-07-30 00:04:48.511029517 -0400
++++ mixxx-2.0.0/src/library/trackcollection.h	2017-07-30 00:05:03.378699826 -0400
+@@ -34,8 +34,7 @@
+ #include "library/dao/libraryhashdao.h"
+
+ #ifdef __SQLITE3__
+-typedef struct sqlite3_context sqlite3_context;
+-typedef struct Mem sqlite3_value;
++#include <sqlite3.h>
+ #endif
+
+ class TrackInfoObject;

